extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeStreamInfoItemExtractor.java

```diff
@@ -284,67 +292,86 @@ public long getViewCount() throws ParsingException {
             try {
-                return Long.parseLong(Utils.removeNonDigitCharacters(
-                        // This approach is language dependent
-                        Parser.matchGroup1("([\\d,]+) views$",
-                                videoInfo.getObject("title")
-                                        .getObject("accessibility")
-                                        .getObject("accessibilityData")
-                                        .getString("label", ""))));
+                return getViewCountFromAccessibilityData();
             } catch (final Exception ignored) {
-                // Ignore all exceptions, as the view count can be hidden by creators, and so
-                // cannot be found in this case
             }
         }
 
         // Fallback to a short view count, always used for livestreams (see why above)
-        try {
+        if (videoInfo.has("videoInfo")) {
             // Returned in playlists, in the form: view count separator upload date
-            if (videoInfo.has("videoInfo")) {
-                return Utils.mixedNumberWordToLong(videoInfo.getObject("videoInfo")
+            try {
+                return getViewCountFromViewCountText(videoInfo.getObject("videoInfo")
                         .getArray("runs")
                         .getObject(0)
-                        .getString("text"));
+                        .getString("text", ""), true);
+            } catch (final Exception ignored) {
             }
+        }
 
+        if (videoInfo.has("shortViewCountText")) {
             // Returned everywhere but in playlists, used by the website to show view counts
-            if (videoInfo.has("shortViewCountText")) {
-                return Utils.mixedNumberWordToLong(videoInfo.getObject("shortViewCountText")
-                        .getArray("runs")
-                        .getObject(0)
-                        .getString("text"));
+            try {
+                final String shortViewCountText =
+                        getTextFromObject(videoInfo.getObject("shortViewCountText"));
+                if (!isNullOrEmpty(shortViewCountText)) {
+                    return getViewCountFromViewCountText(shortViewCountText, true);
+                }
+            } catch (final Exception ignored) {
             }
-        } catch (final Exception ignored) {
-            // Ignore all exceptions, as the view count can be hidden by creators, and so cannot be
-            // found in this case
         }
 
         // No view count extracted: return -1, as the view count can be hidden by creators on videos
         return -1;
     }
 
+    private long getViewCountFromViewCountText(@Nonnull final String viewCountText,
+                                               final boolean isMixedNumber)
+            throws NumberFormatException, ParsingException {
+        // These approaches are language dependent
+        if (viewCountText.toLowerCase().contains("no views")) {
```

Comment:
Why didn't you use `ACCESSIBILITY_DATA_NO_VIEWS_REGEX_LOWERCASE` here, too? Or why didn't you use `.contains("no views")` below, too?

Reply:
I didn't use the regex here because the view count text should only contain the `No views` string and `contains` checks should be faster than using a regex.

For the title accessibility data parsing, we can't use this approach, because if a video title contains `no views`, the extractor will think that there is no views on this video, while it may be not the case.

Reply:
Can't you just do `.endsWith("no views")`?